connectors/sources/confluence.py

```diff
@@ -861,13 +878,18 @@ async def fetch_documents(self, api_query):
                 self.confluence_client.host_url,
                 document.get("_links", {}).get("webui", "")[1:],
             )
+            ancestor_title = " ".join(
+                ancestor["title"] for ancestor in document.get("ancestors", [])
+            )
```

Comment:
Why not have `"ancestors"` return an array, since that's what we get from Confluence? Join then array items into a single string makes it much harder to consume these downstream.

Reply:
We don't get an ancestors title array directly from confluence API. it is list contains dictionary having ancestors. I have shared the output [here](https://github.com/elastic/connectors/issues/2378#issuecomment-2097501463).

Reply:
That's fine. So let's trim everything except for title for now, but leave the structure so that if another user requests some other ancestor metadata, we have an easy way to add it. So our output would be like:
```
{
  ...
  "ancestors": [
    {
      "title": "Foo"
    },
    {
      "title": "Bar"
    }
  ],
  ...
}
```